engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/deployment/DeploymentPreparer.scala

```diff
@@ -0,0 +1,87 @@
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.strategy)).modify(maybeStrategy => maybeStrategy.orElse(Some(Deployment.Strategy.Recreate))) andThen
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.replicas)).modify(maybeReplicas => maybeReplicas.orElse(Some(2))) andThen
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.progressDeadlineSeconds)).modify(config.progressDeadlineSeconds.orElse(_)) andThen
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.template.metadata.name)).set(objectName) andThen
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.template.metadata.labels)).modify(_ ++ labels) andThen
+        (templateSpecLens composeLens GenLens[Pod.Spec](_.volumes)).modify(_ ++ List(Volume("configmap", Volume.ConfigMapVolumeSource(configMapId)))) andThen
+        (templateSpecLens composeLens GenLens[Pod.Spec](_.containers)).modify(containers => modifyContainer(containers, config))
+    deploymentLens(k8sDeploymentConfig)
+  }
+
+  private def withDefault[A](defaultValue: A): Iso[Option[A], A] =
+    Iso[Option[A], A](_.getOrElse(defaultValue))(value => if (value == defaultValue) None else Some(value))
+
+  private def setMinReadySeconds(minReadySeconds: Int) = {
+    minReadySeconds match {
+      case 0 => 10
+      case value => value
+    }
+  }
+
+  private def modifyContainer(containers: List[Container], config: K8sDeploymentManagerConfig): List[Container] = {
+    val image = s"${config.dockerImageName}:${config.dockerImageTag}"
+    val runtimeContainer = Container(
+      name = "runtime",
+      image = image,
+      env = List(
+        EnvVar("SCENARIO_FILE", "/data/scenario.json"),
+        EnvVar("CONFIG_FILE", "/opt/nussknacker/conf/application.conf,/data/modelConfig.conf"),
+        // We pass POD_NAME, because there is no option to pass only replica hash which is appended to pod name.
+        // Hash will be extracted on entrypoint side.
+        EnvVar("POD_NAME", FieldRef("metadata.name"))
+      ),
+      volumeMounts = List(
+        Volume.Mount(name = "configmap", mountPath = "/data")
+      ),
+      // used standard AkkaManagement see HealthCheckServerRunner for details
+      readinessProbe = Some(Probe(new HTTPGetAction(Left(8558), path = "/ready"))),
+      livenessProbe = Some(Probe(new HTTPGetAction(Left(8558), path = "/alive")))
+    )
+
+    def modifySingleContainer(value: Container): Container = {
+      val containerLens = GenLens[Container](_.env).modify(_ ++ runtimeContainer.env) andThen
+        GenLens[Container](_.volumeMounts).modify(_ ++ runtimeContainer.volumeMounts) andThen
+        GenLens[Container](_.readinessProbe).modify(_.orElse(runtimeContainer.readinessProbe)) andThen
+        GenLens[Container](_.livenessProbe).modify(_.orElse(runtimeContainer.livenessProbe))
+      containerLens(value)
+    }
+
+    val (runtimeContainers, userConfiguredContainers) = containers.partition(_.name == "runtime")
```

Comment:
Maybe 
```
containers.map {
  case value if value.name == "runtime" => modifySingleContainer(value)
  case value => value
}
``` 
would be more readable?

Reply:
I want to add "runtime" container if missing, its hard to detect it this way

Reply:
aww, I forgot it's the main reason :D